src/fpu/fpu_instructions.h

```diff
@@ -102,9 +103,29 @@ static void FPU_FPOP(void){
 static double FROUND(double in){
 	switch(fpu.round){
 	case ROUND_Nearest: return std::nearbyint(in);
-	case ROUND_Down: return (floor(in));
-	case ROUND_Up: return (ceil(in));
-	case ROUND_Chop: [[fallthrough]]; // cast by the caller chops
+	case ROUND_Down: return std::floor(in);
+	case ROUND_Up: return std::ceil(in);
+	case ROUND_Chop: 
+	{
+		// This is a fix for rounding to a close integer in
+		// extended precision mode, e.g. 7.999999999999994; an example
+		// can be seen in the Quake options screen size slider.
+		// In this case, what is almost certainly wanted is 8.0,
+		// so return the closer integer within the tolerance instead
+		// of chopping to the lower value.
+		if (fpu.cw & ExtendedPrecisionModeMask) {
+			constexpr double tolerance = 1e-14;
+			const auto lower           = std::floor(in);
+			const auto upper           = std::ceil(in);
+
+			if ((in - lower) < tolerance) {
+				return lower;
+			} else if ((upper - in) < tolerance) {
+				return upper;
+			}
```

Comment:
(just a minor micro-logic comment):
- `upper` isn't needed until we fail the first `lower` check,
- and both `upper` and `lower` aren't needed once the if/else fails

So could be a nice use of C++17's if/else initializer (moving `lower` and `upper` in as initializers).